src/node_url.cc

```diff
@@ -125,6 +106,71 @@ enum url_error_cb_args {
 #undef XX
 };
 
+#define CHAR_TEMPLATE(T, n)                                                   \
+template <typename T,                                                       \
+          typename = typename std::enable_if<std::is_integral<T>::value &&  \
+                                             sizeof(T) >= n / 8, T>::type>
+
```

Comment:
I'd still prefer to have some basic type-checking, but I agree the `is_integral` check is probably overkill. Opted for a `static_assert` on `sizeof(T)` instead.